Add details to documentation on how to deploy to OpenShift and test the application

jackhu008 opened the pull request:
Added details on how to deploy to OpenShift using s2i and test.
Relates to: #11730

jackhu008 committed:
Updated deploying-to-openshift.adoc with details for how to build and
deploy.

geoand commented on docs/src/main/asciidoc/deploying-to-openshift.adoc:
```diff
@@ -56,8 +57,15 @@ By adding this dependency, we now have the ability to configure the OpenShift re
 Furthermore, the items provided by `io.quarkus.deployment.ApplicationConfig` affect the OpenShift resources.
 
-=== Building
+=== Build and deply (in separate steps)
```
```suggestion
=== Build and Deploy (in separate steps)
```

jackhu008 replied:
Thanks. Done.

geoand commented on docs/src/main/asciidoc/deploying-to-openshift.adoc:
```diff
@@ -56,8 +57,15 @@ By adding this dependency, we now have the ability to configure the OpenShift re
-=== Building
+=== Build and deply (in separate steps)
 
+If OpenShift extension was not included during the bootstraping of the project nor added subsequently  (check pom.xml file for it), then add OpenShift as a dependency before building:
```
```suggestion
If the OpenShift extension was not included during the bootstraping of the project nor was it added subsequently  (check pom.xml file for it), then it can be added like so:
```

jackhu008 replied:
Thanks. Done.

geoand commented on docs/src/main/asciidoc/deploying-to-openshift.adoc:
```diff
@@ -90,19 +113,44 @@ or
+=== Build and Deploy (in a single step)
 
-To trigger a deployment:
+If OpenShift extension was not included during the bootstraping of the project nor added subsequently  (check pom.xml file for it), then add OpenShift as a dependency before building:
```
```suggestion
If the OpenShift extension was not included during the bootstraping of the project nor was it added subsequently  (check pom.xml file for it), then it can be added like so:
```

jackhu008 replied:
Thanks. Done.

geoand commented on docs/src/main/asciidoc/deploying-to-openshift.adoc:
```diff
@@ -90,19 +113,44 @@ or
 ----
 
-The command above will trigger a container image build and will apply the generated OpenShift resources, right after.
+The command above will build a jar file locally, subsequently trigger a container image build and then apply the generated OpenShift resources, right after.
```
```suggestion
The aforementioned command will build a jar file locally, trigger a container image build and then apply the generated OpenShift resources.
```

jackhu008 replied:
Thanks. Done.

geoand reviewed (changes requested):
Thanks for this!

I added some suggestions

geoand commented on docs/src/main/asciidoc/deploying-to-openshift.adoc:
```diff
@@ -19,6 +19,7 @@ To complete this guide, you need:
 * JDK 1.8+ installed with `JAVA_HOME` configured appropriately
 * Apache Maven {maven-version}
 * access to an OpenShift cluster (Minishift is a viable option)
+* OpenShift CLI
```
We should probably mention that this is optional

jackhu008 replied:
OpenShift CLI is required for running oc commands from a terminal.  A lot of tasks, not all,  could be achieved using the Web Console instead of using CLI but a developer needs to be comfortable with using CLI in my view.

geoand replied:
Yeah I know, my point is that the CLI is only needed when you want to do perform the manual deployment

jackhu008 replied:
Understood. I have updated line #22 to as follows:

* OpenShift CLI (Optional. Only required for manually deploying)

geoand replied:
Looks great1

geoand commented:
Can you please squash the commits?

jackhu008 commented:
> Can you please squash the commits?

Hi Georgios,  is this a question for me?  If so, how do I squash multiple commits?
I edited the document directly online in my folk and "deploy-to-openshift" branch.  As a result there have been many commits which make the history look longer than it could be.

geoand commented:
Yeah, we try to keep the history clean.

Unfortunately the UI won't let you (or us) squash them, so you'll have to use the command line, see for example: https://www.internalpointers.com/post/squash-commits-into-one-git

jackhu008 commented:
Hi Georgios, thanks for the link.  Will take a look and try to squash commits to one. Jack

jackhu008 commented:
Hi Georgios, I squashed my commits (5) into a single commit. Please check.
Details of what I did: 
`git clone --single-branch --branch deploy-to-openshift https://github.com/jackhu008/quarkus.git`
Then followed the instructions in https://www.internalpointers.com/post/squash-commits-into-one-git .
Finally did a `git pull` and `git push` .  git pull is required here.  Otherwise git push wouldn't work.

geoand commented:
Unfortunately it seems like there are 7 commits now :)

jackhu008 commented:
Odd.  There were 7 commits.  I have squashed them into a single commit.  Have you tried to refresh your browser?

On my browser, the history shows there has been only a single commit from me today since your last commit on 24th August.

https://github.com/jackhu008/quarkus/commits/deploy-to-openshift/docs/src/main/asciidoc/deploying-to-openshift.adoc

geoand commented:
![Screenshot from 2020-09-28 18-04-58](https://user-images.githubusercontent.com/4374975/94449800-29631600-01b5-11eb-8114-9f3aa3956ead.png)

jackhu008 commented:
Hi Georgios, my source repository has squashed the previous 5 commits into a single one. It looks like the the commit history in this PR is not consolidated. Not sure whether it would even get consolidated.  

If I were to raise a new PR now, the new PR would probably only show a single commit, I guess.

geoand commented:
OK, no problem

jackhu008 commented:
Please let me know if you want me to raise a new PR to be double sure.

geoand commented:
Well if squashing doesn't work properly for you, then go ahead and open a new PR

jackhu008 commented:
Tried to raise a new PR. Unfortunately, although my deploy-to-openshift branch commit history only shows one (squashed) commit, the new PR shows 7 commits - same as this PR.  Puzzled... 

Need to look into squash commits further.

If still no luck after further looking into squash commits, I would (as a last resort) create a new branch, and migrate the document changes manually to the new branch in one commit and then raise a PR.

jackhu008 commented:
Hi Georgios, I think I have fixed the squash commits.  I re-ran `git rebase -i HEAD~6` and kept the last consolidated commit and dropped all 5 previous intermediate commits. Finally did a forced push `git push --force-with-lease origin`.  This seem to have got rid of the obsolete intermediate commits. Maybe I should have used forced push when I first time tried squashing commits. 

Please review again - hopefully it is all good for merging now.

geoand reviewed (approved):
Thanks a lot!

jackhu008 commented:
Great!  
Just curious, what's the remaining process for the changes to be reflected in the online documentation https://quarkus.io/guides/deploying-to-openshift?  And how long would that take?

geoand commented:
They will appear when we do a new 1.8.x release or when 1.9.x comes along

nheidloff commented:
I like this. But I'd correct the spelling of the title to "OpenShift"?

geoand commented:
Thanks for spotting it @nheidloff

jackhu008 commented:
> I like this. But I'd correct the spelling of the title to "OpenShift"?

Oops!  I knew one day I might embarrass myself with an incorrect spelling of "OpenShift" but did think it would be so publically done.  Thanks for spotting it @nheidloff and for correcting @geoand .
